umap/static/umap/js/umap.forms.js

```diff
@@ -802,20 +804,116 @@ L.FormBuilder.FacetSearch = L.FormBuilder.Element.extend({
     input.dataset.value = value
-    label.htmlFor = `checkbox_${this.name}_${value}`
+    label.htmlFor = `${this.inputType}_${this.name}_${value}`
     label.innerHTML = value
+
     L.DomEvent.on(input, 'change', (e) => this.sync())
   },
 
   toJS: function () {
-    return [...this.ul.querySelectorAll('input:checked')].map((i) => i.dataset.value)
+    return {
+      'inputType': this.inputType,
+      'choices': [...this.ul.querySelectorAll('input:checked')].map((i) => i.dataset.value)
+    }
   },
 })
 
+L.FormBuilder.FacetSearchMinMax = L.FormBuilder.Element.extend({
+  build: function () {
+    this.container = L.DomUtil.create('div', 'umap-facet', this.parentNode)
+    this.table = L.DomUtil.create('table', '', this.container)
+    this.inputType = this.options.criteria["inputType"]
+
+    const min = this.options.criteria['min']
+    const max = this.options.criteria['max']
+
+    this.minTr = L.DomUtil.create('tr', '', this.table)
+
+    this.minTdLabel = L.DomUtil.create('td', '', this.minTr)
+    this.minLabel = L.DomUtil.create('label', '', this.minTdLabel)
+    this.minLabel.innerHTML = 'Min'
+    this.minLabel.htmlFor = `${this.inputType}_${this.name}_min`
+
+    this.minTdInput = L.DomUtil.create('td', '', this.minTr)
+    this.minInput = L.DomUtil.create('input', '', this.minTdInput)
+    this.minInput.type = this.inputType
+    this.minInput.id = `${this.inputType}_${this.name}_min`
+    this.minInput.step = '1'
+    if (min != null) {
+      this.minInput.valueAsNumber = min.valueOf()
+      this.minInput.dataset.value = min
+    }
+
+    this.maxTr = L.DomUtil.create('tr', '', this.table)
+
+    this.maxTdLabel = L.DomUtil.create('td', '', this.maxTr)
+    this.maxLabel = L.DomUtil.create('label', '', this.maxTdLabel)
+    this.maxLabel.innerHTML = 'Max'
+    this.maxLabel.htmlFor = `${this.inputType}_${this.name}_max`
+
+    this.maxTdInput = L.DomUtil.create('td', '', this.maxTr)
+    this.maxInput = L.DomUtil.create('input', '', this.maxTdInput)
+    this.maxInput.type = this.inputType
+    this.maxInput.id = `${this.inputType}_${this.name}_max`
+    this.maxInput.step = '1'
+    if (max != null) {
+      this.maxInput.valueAsNumber = max.valueOf()
+      this.maxInput.dataset.value = max
+    }
+
+    if (["date", "datetime-local"].includes(this.inputType)) {
+      this.minLabel.innerHTML = 'From'
+      this.maxLabel.innerHTML = 'Until'
+      if (min != null) {
+        this.minInput.valueAsNumber = (min.valueOf() - min.getTimezoneOffset() * 60000)
+      }
+      if (max != null) {
+        this.maxInput.valueAsNumber = (max.valueOf() - max.getTimezoneOffset() * 60000)
+      }
+    }
+
+    if (["datetime-local"].includes(this.inputType)) {
+      this.minInput.step = '0.001'
+      this.maxInput.step = '0.001'
+    }
+
+    if (["number"].includes(this.inputType)) {
+      if (min != null && max != null) {
+        // calculate step from significant digits of min and max values
```

Comment:
This should be extracted in a small util (in umap.core.js) and tested :)

Reply:
Can you recheck this one, please? I moved the functionality to umap.core.js as you suggested.